apps/checkin/views.py

```diff
@@ -1,56 +1,163 @@
 from .models import CheckInGroup, CheckInEvent
 import qrcode
 from django.utils import timezone
-from .models import User
 from ..application.models import Application
 from ..constants.models import CONSTANTS
 
 from rest_framework.permissions import IsAuthenticated, IsAdminUser
 from rest_framework.views import APIView
 from rest_framework.response import Response
+from rest_framework.decorators import api_view
 import base64
 from io import BytesIO
+from ..events.views import error_response, success_data_jsonify
+from ..registration.models import User
+from django.contrib.auth import authenticate
+import jwt
+from madras.settings import SECRET_KEY
+import json
+from django.http import HttpResponse
+from wsgiref.util import FileWrapper
+from wallet.models import Pass, Barcode, EventTicket, Location, BarcodeFormat
+import os
+import hashlib
 
-static_path = "static/checkin/qr_codes/"
 
+@api_view(['POST'])
+def get_qr_code(request):
+    params = json.loads(request.body.decode("utf-8"))
+    if "username" not in params:
+        return error_response("You must include your username in the request.",
+                              400)
+    if "password" not in params:
+        return error_response("You must include your password in the request",
+                              400)
+    user = authenticate(username=params["username"],
+                        password=params["password"])
+    if user is None:
+        return error_response("Invalid Login Credentials", 401)
+    try:
+        application = Application.objects.get(user=user)
+    except Application.DoesNotExist:
+        return error_response(
+            "You do not have an application.",
+            404,
+        )
+    if not CONSTANTS.objects.get().DECISIONS_RELEASED:
+        return error_response(
+            "Decisions have not been released.",
+            403,
+        )
+    if application.admission_status != "A":
```

Comment:
this is sketch af.

Reply:
explain, why is this sketch af.

Reply:
admission_status should not be compared to the letter 'A'. There should be like an enum for this...not really your fault.

Reply:
1) It is my fault, because I wrote the field ;)
2) John deleted admission_status anyway and I made a task for him to add it back somewhere.

Reply:
Just a note, it doesn't look like the `admission_status` field exists in the `Application` model anymore.